src/main/java/org/axonframework/springboot/nativex/MessageHandlerRuntimeHintsRegistrar.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.springboot.nativex;
+
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.annotation.AnnotatedHandlerInspector;
+import org.axonframework.messaging.annotation.MessageHandlingMember;
+import org.axonframework.queryhandling.annotation.QueryHandlingMember;
+import org.axonframework.spring.config.MessageHandlerLookup;
+import org.springframework.aot.generate.GenerationContext;
+import org.springframework.aot.hint.BindingReflectionHintsRegistrar;
+import org.springframework.aot.hint.ExecutableMode;
+import org.springframework.aot.hint.ReflectionHints;
+import org.springframework.beans.factory.aot.BeanFactoryInitializationAotContribution;
+import org.springframework.beans.factory.aot.BeanFactoryInitializationAotProcessor;
+import org.springframework.beans.factory.aot.BeanFactoryInitializationCode;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * BeanFactoryInitializationAotProcessor that registers message handler methods declared on beans for reflection. This
+ * means that all methods annotated with {@code @MessageHandler} will be available for reflection.
+ * <p/>
+ * Additionally, the payload types for these methods are registered for reflective access, as well as the classes
+ * containing the methods.
+ *
+ * @author Allard Buijze
+ * @since 4.8.0
+ */
+public class MessageHandlerRuntimeHintsRegistrar implements BeanFactoryInitializationAotProcessor {
+
+    @Override
+    public BeanFactoryInitializationAotContribution processAheadOfTime(ConfigurableListableBeanFactory beanFactory) {
+        List<Class<?>> messageHandlingClasses =
+                MessageHandlerLookup.messageHandlerBeans(messageType(), beanFactory, true)
```

Comment:
Since we use the `MessageHandlerLookup` in Axon Framework internally to find all bean names, that means we're retrieving all handlers at least twice, right?
Nuding @abuijze directly, as this is his introduction to this PR.

Reply:
I'm not sure why we retrieve it twice. I broke up the original but, so we can also set reflection hints on the classes, so we could cleanup the icky find classes in AxonNativeHints. It re-uses the list to find the handlers.

Reply:
To be clear, I am talking about Axon Framework's `InfraConfiguration` in the auto-configuration module that constructs the `MessageHandlerLookup` bean.
This bean, as its `BeanDefinitionRegistryPostProcessor` implementation, uses the `MessageHandlerLookup#messageHandlerBeans` operation during the `BeanFactoryPostProcessor#postProcessBeanFactory`.
This operation, in turn, constructs the `MessageHandlerConfigurer` to register command, event, and query handling components with their respective buses.

So, this addition means that whenever users combine Axon Framework with our Spring AOT extension, that `MessageHandlerLookup#messageHandlerBeans` is invoked twice.
I don't think that's something we can resolve here, though. Just wanted to call it out as it _should_ prolong startup time.